### PR TITLE
extraction-2-init-reusable-transforms-and-setup-test-utils - CTR-20

### PR DESCRIPTION
**Pull request by kiprasmel**
how to review: https://github.com/pipedrive/CodeshiftCommunity/pull/9 (commit-by-commit is still best)

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "esModuleInterop": true,
     "allowJs": true,
     "paths": {
-      "@codeshift/*": ["../packages/*/src"]
+      "@codeshift/*": ["../packages/*/src", "../packages/reusable-transforms"]
```
Comment (martikaljuve) [inline tsconfig.json:24]: Probably unnecessary?

File: community/@pipedrive__convention-ui-react/src/test-utils/inlineTest.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+export type InlineTestPropsA<C> = InlineTestPropsBase<C> & {
+    from: string;
+    to: string;
+};
+export type InlineTestPropsB<C> = InlineTestPropsBase<C> & {
+    fromTo: string;
+};
+export type InlineTestProps<C> = InlineTestPropsA<C> | InlineTestPropsB<C>;
+
+export type InlinteTestRet = ReturnType<typeof defineInlineTest>;
```
Comment (martikaljuve) [inline community/@pipedrive__convention-ui-react/src/test-utils/inlineTest.ts:29]:
```suggestion
export type InlineTestRet = ReturnType<typeof defineInlineTest>;
```

File: community/@pipedrive__convention-ui-react/src/test-utils/inlineTest.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+             * lmao, atm only `flow` works for both js & ts
+             *
+             * in the astexplorer, the `@typescript-eslint/parser` worked well,
+             * I wonder if we can add it (probably yes).
+             *
+             * TODO test more scenarios of different langs we're using
+             * before releasing to make sure we don't screw up
+             * for a common scenario.
+             *
+             * though JS & TS already deep in what I wanna support.
```
Comment (martikaljuve) [inline community/@pipedrive__convention-ui-react/src/test-utils/inlineTest.ts:47]: update?

**Review by martikaljuve — APPROVED** [review]

**Review by martikaljuve — CHANGES_REQUESTED** [review]
.

File: CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,2 @@
+
+/packages/reusable-transforms/ @kiprasmel
```
Comment (kiprasmel) [inline CODEOWNERS:2]: fyi, this is because i want to be a codeowner on upstream.

File: tsconfig.json (hunk 1 above)
Comment (kiprasmel) [reply tsconfig.json]: fixed

File: community/@pipedrive__convention-ui-react/src/test-utils/inlineTest.ts (hunk 3 above)
Comment (kiprasmel) [reply community/@pipedrive__convention-ui-react/src/test-utils/inlineTest.ts]: updated.

File: packages/reusable-transforms/src/utils/Transformer.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+// `unknown` not enough -- `any` intentionally
+export type TupleOfTransformerAndConfigs<Config = any> = readonly [Transformer<Config>, Config[]] // TODO ESLINT
+
+export const createCodemodRunnerFromMultiple = <
+    Tuples extends readonly TupleOfTransformerAndConfigs[] = TupleOfTransformerAndConfigs[],
+>(
+    tuplesOfTransformsWithTypesafeConfigs: Tuples,
+) => (
+	...transformerParamsExceptConfig: [j: JSCodeshift, src: Collection<any>, fileInfo: FileInfo, api: API, options: Options]
```
Comment (martikaljuve) [inline packages/reusable-transforms/src/utils/Transformer.ts:20]: I think these named tuples are not supported by the old (1.x) version of Prettier that this project is using. Could we upgrade Prettier to 2.x (latest seems to be 2.5.1)?

**Review by martikaljuve — APPROVED** [review]

**Pull request merged**